dev.md uses -Dwarning_level=3 but CI doesnt; documented dev build fails

Following the "Hack nixd from source" section in `nixd/docs/dev.md`:

```
meson setup build --buildtype=debug --default-library=static --prefix=$PWD/local/install -Dwarning_level=3 -Dwerror=true
meson compile -C build
```

This fails on every use of CheckReturn/CheckDefault:

```
CheckReturn.h:7:3: error: ISO C++ forbids braced-groups within expressions [-Werror=pedantic]
```

From my investigation, its caused by a combination of the following:
- nix's pkg-config injects `-std=c++23` into cflags (`c++20` fails identically the break is `gnu++` vs `c++`, not 20 to 23)
- meson appends dependency cflags after the project's `cpp_std=gnu++20` (last `-std` wins, so the effective dialect is strict ISO C++23, not gnu++20)
- `CheckReturn.h` uses a [statement-expression](https://gcc.gnu.org/onlinedocs/gcc/Statement-Exprs.html) and strict ISO forbids them, `-Wpedantic` flags them, `-Dwerror=true` makes it fatal

`.github/workflows/build.yml` has `-Dwerror=true` but not `-Dwarning_level=3`, so CI uses the default warning level 2 where `-Wpedantic` is absent. The dev docs uses a stricter config than what CI uses.

Should `warning_level=3` be a supported build?
- **yes**: mark the statement-expr with `__extension__` and add `-Dwarning_level=3` to the CI matrix so it cant regress
- **no**: drop `-Dwarning_level=3` from `dev.md` to match CI

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -24,7 +24,7 @@ jobs:
     - uses: cachix/install-nix-action@v31
       with:
         github_access_token: ${{ secrets.GITHUB_TOKEN }}
-    - run: nix develop --command meson setup build/ -Dwerror=true --buildtype=${{ matrix.buildtype }} -Db_sanitize=${{ matrix.sanitizer }} -Db_ndebug=${{ matrix.ndebug }}
+    - run: nix develop --command meson setup build/ -Dwerror=true -Dwarning_level=3 --buildtype=${{ matrix.buildtype }} -Db_sanitize=${{ matrix.sanitizer }} -Db_ndebug=${{ matrix.ndebug }}
     - run: nix develop --command meson compile -C build
     - run: nix develop --command meson test -C build
     - uses: actions/upload-artifact@v7
@@ -49,7 +49,7 @@ jobs:
     - uses: cachix/install-nix-action@v31
       with:
         github_access_token: ${{ secrets.GITHUB_TOKEN }}
-    - run: nix develop .#llvm --command env CXX=clang++ meson setup build/ -Dwerror=true --buildtype=${{ matrix.buildtype }} -Db_sanitize=${{ matrix.sanitizer }} -Db_ndebug=${{ matrix.ndebug }}
+    - run: nix develop .#llvm --command env CXX=clang++ meson setup build/ -Dwerror=true -Dwarning_level=3 --buildtype=${{ matrix.buildtype }} -Db_sanitize=${{ matrix.sanitizer }} -Db_ndebug=${{ matrix.ndebug }}
     - run: nix develop .#llvm --command meson compile -C build
     - run: nix develop .#llvm --command meson test -C build
     - uses: actions/upload-artifact@v7

